### PR TITLE
Added benchmarking targets.

### PR DESCRIPTION
**Pull request by mofarrell**
Also added a kernel benchmark.

### CODE REVIEW

File: osquery/main/benchmarks.cpp [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+DECLARE_string(database_path);
+DECLARE_string(extensions_socket);
+DECLARE_string(modules_autoload);
+DECLARE_string(extensions_autoload);
+DECLARE_bool(disable_logging);
+
+typedef std::chrono::high_resolution_clock chrono_clock;
+
+void initTesting() {
```
Comment (theopolis) [inline osquery/main/benchmarks.cpp]: Could we extern or add this to `core/test_util.{h,cpp}`?

File: osquery/main/benchmarks.cpp [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  FLAGS_disable_logging = true;
+
+  // Create a default DBHandle instance before unittests.
+  (void)DBHandle::getInstance();
+}
+}
+
+int main(int argc, const char* argv[]) {
+  // Allow unit test execution from anywhere in the osquery source/build tree.
+  while (osquery::kTestDataPath != "/") {
```
Comment (theopolis) [inline osquery/main/benchmarks.cpp]: This logic might be better in `initTesting`.

File: tools/benchmark.sh [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+trap cleanUp EXIT INT TERM
+
+# Load osquery kernel (optional).
+make kernel-load
+
+OUTDIR="$SCRIPT_DIR/../build/benchmarks"
+mkdir -p $OUTDIR
+
+export BENCHMARK_TO_FILE="--benchmark_format=csv :>$OUTDIR/benchmark.csv"
+make run-benchmark
```
Comment (theopolis) [inline tools/benchmark.sh]: Do the benchmarks add a lot of time to the builds? Could we run benchmarks on EVERY PR and merge into master?

File: osquery/main/benchmarks.cpp (hunk 2 above)
Comment (mofarrell) [inline osquery/main/benchmarks.cpp]: Ok I was basically copying the structure of the `main/tests.cpp` file

File: osquery/main/benchmarks.cpp (hunk 1 above)
Comment (mofarrell) [inline osquery/main/benchmarks.cpp]: Sure and make it common between `main/tests.cpp` and here

File: kernel/src/osquery.cpp [hunk 4]
```diff
@@ -254,13 +257,7 @@ static int osquery_ioctl(dev_t dev, u_long cmd, caddr_t data,
...
     *(int *)e = test_counter;
     char *s = (char *)((int *)e + 1);
-    s[0] = 'H';
-    s[1] = 'E';
-    s[2] = 'L';
-    s[3] = 'L';
-    s[4] = 'O';
-    s[5] = '!';
-    s[6] = '\0';
+    memset(s, 'H', length);
```
Comment (theopolis) [inline kernel/src/osquery.cpp]: Real hackers use a [nopseld](https://www.corelan.be/index.php/2011/12/31/exploit-writing-tutorial-part-11-heap-spraying-demystified/#0x0c0c0c0c). Just saying.

File: CMakeLists.txt [hunk 5]
```diff
@@ -377,3 +378,10 @@ add_custom_target(
   COMMENT "Generating sdk sync: ${CMAKE_BINARY_DIR}/sync"
   DEPENDS osquery_extensions osquery_amalgamation
 )
+
+# make benchmark
+add_custom_target(
+  run-benchmark
```
Comment (theopolis) [inline CMakeLists.txt]: Maybe just `make benchmark` and ditch the run?

Comment (mofarrell) [inline CMakeLists.txt]: The benchmark library uses that target.  Is there an easy way around that one?

File: tools/benchmark.sh (hunk 3 above)
Comment (mofarrell) [inline tools/benchmark.sh]: Every benchmark is repeated for as many iterations as possible until hitting a time cap and then it calculates time per iteration.  So I would say they take a little while.  I believe every benchmark will run about a half second.  Here the kernel communication benchmark I added probably take 4 seconds or so.

File: CMakeLists.txt (hunk 5 above)
Comment (theopolis) [inline CMakeLists.txt]: Ah, that's ubfortunate. "Run" will have to suffice.

**Comment by mofarrell** [comment]
ok to test

**Comment by mofarrell** [comment]
ok to test

**Comment by mofarrell** [comment]
ok to test

**Comment by mofarrell** [comment]
ok to test

**Pull request merged**